ServerSide/yoty/Core/Managers/Bids/BidsManager.cs

```diff
@@ -0,0 +1,291 @@
+                }
+            }
+            BidDTO dto = null;
+            //TODO map ent to dto
+            return new Response<BidDTO>() { DTOObject = dto, IsOperationSucceeded = true, SuccessOrFailureMessage = this.getSuccessMessage() };
+
+        }
+
+        public async Task<Response> DeleteBid(string bidId)
+        {
+            var bid = await _context.Bids.FindAsync(bidId).ConfigureAwait(false);
+            if (bid == null)
+            {
+                return new Response() { IsOperationSucceeded = false, SuccessOrFailureMessage = BidNotFoundFailString };
+            }
+            using (var new_context = new YotyContext())
+            {
+                new_context.Bids.Remove(bid);
+                try
+                {
+                    await new_context.SaveChangesAsync().ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    return new Response() { IsOperationSucceeded = false, SuccessOrFailureMessage = ex.Message };
+                }
+            }
+            return new Response() { IsOperationSucceeded = true, SuccessOrFailureMessage = this.getSuccessMessage() };
+        }
+
+        public async Task<Response> DeleteBuyer(string bidId, string buyerId)
+        {
+            BidEntity bid = await _context.Bids.FindAsync(bidId).ConfigureAwait(false);
+            if (bid == null)
+            {
+                return new Response() { IsOperationSucceeded = false, SuccessOrFailureMessage = BidNotFoundFailString };
+            }
+            ParticipancyEntity participancy = bid.CurrentParticipancies.Find(p => p.BuyerId == buyerId);
+            bid.UnitsCounter -= participancy.NumOfUnits;
+            bid.CurrentParticipancies.Remove(participancy);
+            using (var new_context = new YotyContext())
+            {
+                try
+                {
+                    new_context.Bids.Update(bid);
+                    await new_context.SaveChangesAsync().ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    return new Response() { IsOperationSucceeded = false, SuccessOrFailureMessage = ex.Message };
+                }
+            }
+            return new Response() { IsOperationSucceeded = true, SuccessOrFailureMessage = this.getSuccessMessage() };
+        }
+
+        public async Task<Response> DeleteSupplierProposal(string bidId, string supplierId)
+        {
+            BidEntity bid = await _context.Bids.FindAsync(bidId).ConfigureAwait(false);
+            if (bid == null)
+            {
+                return new Response() {IsOperationSucceeded = false, SuccessOrFailureMessage = BidNotFoundFailString };
+            }
+            SellerOfferEntity proposal = bid.CurrentOffers.Find(p => p.SellerId == supplierId);
+            bid.CurrentOffers.Remove(proposal);
+            bid.PotenialSuplliersCounter--;
```

Comment:
I think that we have one unnecessary access to the bid table .... the CurrentOffers of the bid is actually sub set of the SellerOffer table, so lets access directly to SellerOffer table and delete it